### PR TITLE
Typo fix

### PR DESCRIPTION
**Pull request by janekl**
I'm fixing an issue with choosing distribution type within `Multihot` class: it uses `type` variable which shadows Python built-in function which, in turn, shadows a problem with comparisons like `type == "uniform"` in `__make_multi_hot_indices_cache` method.

### CODE REVIEW

**Review by samiwilf — APPROVED** [review]
Thanks Jan for catching this. LGTM.  Side note, this issue wasn't in the code I ran to report 20-hot pareto distribution AUROC results of 0.7957 and 0.7958 for validation and test sets.  To double check, I ran dlrm_main.py again with this PR applied, and the AUROC results after 1 epoch were 0.7960 for both validation and test sets.   The exact command I ran on a p4d.24xlarge: 
torchx run -s local_cwd dist.ddp -j 1x8 --script dlrm_main.py -- --pin_memory --batch_size 2048 --epochs 1 --num_embeddings_per_feature "45833188,36746,17245,7413,20243,3,7114,1441,62,29275261,1572176,345138,10,2209,11267,128,4,974,14,48937457,11316796,40094537,452104,12606,104,35" --embedding_dim 128 --dense_arch_layer_sizes "512,256,128" --over_arch_layer_sizes "1024,1024,512,256,1" --in_memory_binary_criteo_path "/home/ubuntu/mountpoint/shuffled" --learning_rate 1.0 --mmap_mode --shuffle_batches --multi_hot_size 20 --multi_hot_distribution_type "pareto" --validation_freq_within_epoch 30000
